SMLHelper/Json/ConfigFile.cs

```diff
@@ -0,0 +1,90 @@
+﻿namespace SMLHelper.V2.Json
+{
+    using System.IO;
+    using System.Reflection;
+    using System.Linq;
+    using SMLHelper.V2.Json.Converters;
+    using SMLHelper.V2.Json.ExtensionMethods;
+    using SMLHelper.V2.Json.Interfaces;
+    using Oculus.Newtonsoft.Json;
+    using Oculus.Newtonsoft.Json.Converters;
+
+    /// <summary>
+    /// A simple implementation of <see cref="IJsonFile"/> for use with config files.
+    /// </summary>
+    public abstract class ConfigFile : IJsonFile
+    {
+        /// <summary>
+        /// The file path at which the JSON file is accessible for reading and writing.
+        /// </summary>
+        public string JsonFilePath { get; private set; }
+
+        [JsonIgnore]
+        private readonly JsonConverter[] alwaysIncludedJsonConverters = new JsonConverter[] {
+            new KeyCodeConverter(),
+            new FloatConverter(),
+            new StringEnumConverter(),
+            new VersionConverter()
+        };
+
+        /// <summary>
+        /// The <see cref="JsonConverter"/>s that should always be used when reading/writing JSON data.
+        /// </summary>
+        /// <seealso cref="alwaysIncludedJsonConverters"/>
+        public JsonConverter[] AlwaysIncludedJsonConverters => alwaysIncludedJsonConverters;
```

Comment:
Also, that can be marked as virtual, so in case user needs to add some custom converters they can be added in the overrided property.

Reply:
The way I see this, there should actually be 2 properties.

One `internal` property that returns the `alwaysIncludedJsonConverters`,  
and a second `public` property that the modder can set or override with any additional JsonConverters they wish to use.

Reply:
> modder can set or override with any additional JsonConverters they wish to use.
Or maybe modder wants to use their own custom VersionConverter? :)

Reply:
They can, quite easily. JsonConverters added directly as attributes on properties will take precedence over any other JsonConverters. And if they want to *entirely* exclude a JsonConverter that is in the always included, they can also make their own implementation of IJsonFile, or their own derived class of ConfigFile (yeah, I should make it possible to override the AlwaysIncludedJsonConverters property).

Reply:
The point of ConfigFile class is to have a basic, simple implementation of IJsonFile that covers the majority of use cases for working specifically with a commonly implemented config.json file. Currently, modders are reinventing the wheel every single time they want to do this (or they're reinventing the wheel once for themselves and using a Shared Project or something to use it again and again in any of their mods that use a config.json). What I am attempting to do here is make a simple, standardised API so that people don't *need* to keep reinventing the wheel, that covers the majority of use cases for someone who just wants to load and save properties to a config.json. An API that is simple and easy enough for someone who is making their very first mod to use.

If you want to go deeper with it, that's where you either write your own IJsonFile implementation, or make use of JsonUtils.Save() and JsonUtils.Load() directly. There's _nothing_ to stop people from not using the JSON API provided here at all and use Unity's own, or LitJson, or Newtonsoft.Json directly if they want. This is just to make it so you don't have to think about any of that in the majority of cases.

Reply:
> JsonConverters added directly as attributes on properties will take precedence over any other JsonConverters

Yeah, I mean with that overrided property. You can add them with overrided property, you can add them with attributes and you can add them via method parameters. Maybe simplify that?

Reply:
I do agree it doesn't *need* to be visible as parameters on the Load() and Save() instance methods of ConfigFile or IJsonFile. They're mostly there for greater flexibility of usage - to allow for more use cases- but I could see myself conceding on removing them as parameters from ConfigFile.Load()/Save() instance methods. I absolutely want to keep them on the JsonUtils.Load()/Save() static methods, however.

Reply:
> I absolutely want to keep them on the JsonUtils.Load()/Save() static methods, however.

Maybe just add JsonSerializerSettings instead? I think, it's possible to include converters there.